### PR TITLE
UUID regression testkit

### PR DESCRIPTION
**Pull request by nmartynenko**
Hi @szeiger 

This PR covers:

* add UUID column for roundtrip tests
* a port of #989 for v3.0 (with tests)
* creating unified test for Postgres and H2
* new test showed, that #1058 is wrong (in my defence I must say, that I did tests for H2 in Postgres [compatibility mode](http://www.h2database.com/html/features.html#compatibility))

If this task should be separated in few commits, or something should not be here or else, please tell me, I'll try to fix this.

Basically it is a follow up of #1069.

Cheers,  Nick.

### CODE REVIEW

**Comment by szeiger** [comment]
As noted in #1082, #1058 has not been merged into master yet (I didn't notice that it targeted 2.1-no-bin-compat). Should it still be merged or does this PR supersede it?

**Comment by nmartynenko** [comment]
Hi @szeiger 

It does cover with all appropriate code adjustments.

Cheers, Nick.

**Comment by szeiger** [comment]
OK, great, and thanks for the continued effort on this. LGTM.

**Comment by nmartynenko** [comment]
You're welcome!
